doc/source/ray-air/trainer.rst

```diff
@@ -0,0 +1,215 @@
+
+See :ref:`here for a more detailed user-guide <air-trainers-gbdt-user-guide>`.
+
+
+XGBoost Trainer
+~~~~~~~~~~~~~~~
+
+Ray AIR also provides an easy to use :class:`XGBoostTrainer <ray.train.xgboost.XGBoostTrainer>`
+for training XGBoost models at scale.
+
+To use this trainer, you will need to first run: ``pip install -U xgboost-ray``.
+
+.. literalinclude:: doc_code/xgboost_trainer.py
+    :language: python
+
+LightGBMTrainer
+~~~~~~~~~~~~~~~
+
+Similarly, Ray AIR comes with a :class:`LightGBMTrainer <ray.train.lightgbm.LightGBMTrainer>`
+for training LightGBM models at scale.
+
+To use this trainer, you will need to first run ``pip install -U lightgbm-ray``.
+
+.. literalinclude:: doc_code/lightgbm_trainer.py
+    :language: python
+
+.. _air-trainers-other:
+
+Other Trainers
+--------------
+
+HuggingFace Trainer
+~~~~~~~~~~~~~~~~~~~
+
+:class:`HuggingFaceTrainer <ray.train.huggingface.HuggingFaceTrainer>` further extends :class:`TorchTrainer <ray.train.torch.TorchTrainer>`, built
+for interoperability with the HuggingFace Transformers library.
+
+Users are required to provide a ``trainer_init_per_worker`` function which returns a
+``transformers.Trainer`` object. The ``trainer_init_per_worker`` function
+will have access to preprocessed train and evaluation datasets.
+
+Upon calling `HuggingFaceTrainer.fit()`, multiple workers (ray actors) will be spawned,
+and each worker will create its own copy of a ``transformers.Trainer``.
+
+Each worker will then invoke ``transformers.Trainer.train()``, which will perform distributed
+training via Pytorch DDP.
+
+
+.. dropdown:: Code example
+
+    .. literalinclude:: doc_code/hf_trainer.py
+        :language: python
+        :start-after: __hf_trainer_start__
+        :end-before: __hf_trainer_end__
+
+
+Scikit-Learn Trainer
+~~~~~~~~~~~~~~~~~~~~
+
+.. note:: This trainer is not distributed.
+
+The Scikit-Learn Trainer is a thin wrapper for one to launch scikit-learn training within Ray AIR.
+It is not distributed but can still benefit from integrating with Ray Tune and batch/online prediction.
+
+.. literalinclude:: doc_code/sklearn_trainer.py
+    :language: python
+
+
+RLlib Trainer
+~~~~~~~~~~~~~
+
+RLTrainer provides an interface to RL Trainables. This enables you to use the same abstractions
+as in the other trainers to define the scaling behavior, and to use Ray Data for offline training.
+
+Please note that some scaling behavior still has to be defined separately.
+The :class:`scaling_config <ray.air.config.ScalingConfig>` will set the number of
+training workers ("Rollout workers"). To set the number of e.g. evaluation workers, you will
+have to specify this in the ``config`` parameter of the ``RLTrainer``:
+
+.. literalinclude:: doc_code/rl_trainer.py
+    :language: python
+
+
+How to interpret training results?
+----------------------------------
+
+Calling ``Trainer.fit()`` will return a :class:`Result <ray.air.Result>` object.
+You can interact with a `Result` object as follows:
```

Comment:
Might be better to use present tense than future tense since this is always true.
The ``Trainer.fit()`` call returns a :class:`Result <ray.air.Result>` object, providing you 
access to metrics, checkpoints, error, and best checkpoints.